Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -301,6 +301,61 @@ theorem limitRecOn_limit {C} (o H₁ H₂ H₃ h) :
     @limitRecOn C o H₁ H₂ H₃ = H₃ o h fun x _h => @limitRecOn C x H₁ H₂ H₃ := by
   simp_rw [limitRecOn, SuccOrder.prelimitRecOn_limit _ _ h.isSuccPrelimit, dif_neg h.1]
 
+/-- Bounded recursion on ordinals. Similar to `limitRecOn`, with the assumption `o < l`
+  added to all cases. The final term's domain is the ordinals below `Iio l`. -/
+@[elab_as_elim]
+def boundedLimitRec {l : Ordinal} (hLim : l.IsLimit) {C : Iio l → Sort*} (H₁ : C ⟨0, hLim.pos⟩)
+    (H₂ : (o : Iio l) → C o → C ⟨o + 1, hLim.succ_lt o.2⟩)
```

Comment:
Note that in the current API, `succ o` is the preferred form to `o + 1`. There are however plans to change this in the near future, so I'll leave to you keeping this as `o + 1` or changing it to `succ o`.